src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -170,32 +170,47 @@ private async Task ReportGroupsAsync(ImmutableArray<ISymbol> symbols, Cancellati
 
         private async Task ProcessProjectAsync(Project project, ImmutableArray<ISymbol> allSymbols, CancellationToken cancellationToken)
```

Comment:
this is the major top level algorithmic change.  as we start to search a project, we first see if the symbols we're searching for could have global aliases to them in that project.  We then pass those global aliases along to the actual 'finders' responsible for figuring out what doucments to search and what to search for in those documents.